cpp/models/ode_secirts/infection_state.h

```diff
@@ -0,0 +1,74 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Henrik Zunker, Wadim Koslow, Daniel Abele, Martin J. Kühn
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef MIO_ODE_SECIRTS_INFECTIONSTATE_H
+#define MIO_ODE_SECIRTS_INFECTIONSTATE_H
+namespace mio
+{
+namespace osecirts
+{
+
+/**
+    * @brief The InfectionState enum describes the possible
+    * categories for the infectious state of persons.
+    * Enum is usable as an index, e.g. in CustomIndexArray.
+    * Suffix `Naive` means no immunity through vaccination or infection.
+    * Suffix `PartialImmunity` means vaccinated once.
+    * Suffix `ImprovedImmunity` means vaccinated twice or recovered from infection.
+    * Suffix `Confirmed` means infection has been confirmed by a test, e.g. during commute.
+    */
+enum class InfectionState
+{
+    SusceptibleNaive = 0,
+    SusceptiblePartialImmunity,
+    ExposedNaive,
+    ExposedPartialImmunity,
+    ExposedImprovedImmunity,
+    InfectedNoSymptomsNaive,
+    InfectedNoSymptomsPartialImmunity,
+    InfectedNoSymptomsImprovedImmunity,
+    InfectedNoSymptomsNaiveConfirmed,
+    InfectedNoSymptomsPartialImmunityConfirmed,
+    InfectedNoSymptomsImprovedImmunityConfirmed,
+    InfectedSymptomsNaive,
+    InfectedSymptomsPartialImmunity,
+    InfectedSymptomsImprovedImmunity,
+    InfectedSymptomsNaiveConfirmed,
+    InfectedSymptomsPartialImmunityConfirmed,
+    InfectedSymptomsImprovedImmunityConfirmed,
+    InfectedSevereNaive,
+    InfectedSeverePartialImmunity,
+    InfectedSevereImprovedImmunity,
+    InfectedCriticalNaive,
+    InfectedCriticalPartialImmunity,
+    InfectedCriticalImprovedImmunity,
+    SusceptibleImprovedImmunity,
+    DeadNaive,
+    DeadPartialImmunity,
+    DeadImprovedImmunity,
+    TemporaryImmunPartialImmunity,
+    TemporaryImmunImprovedImmunity,
```

Comment:
```suggestion
    TemporaryImmunePartialImmunity,
    TemporaryImmuneImprovedImmunity,
```

If this was meant to be short for `Immunity`, OK, but it looks a little bit like wrongly written `Immune` while, then, temporary would have been temporarily.. Don't know 100% how to proceed or rename.

Reply:
Initially, `TemporaryImmunPartialImmunity` was used as abbreviation for TemporaryImmunityPartialImmunity. However, i also like the idea to add the 'e' here.